pkg/features/system/collector.go

```diff
@@ -369,3 +386,21 @@ func (c *systemCollector) collectSatelites(client *rpc.Client, ch chan<- prometh
 		ch <- prometheus.MustNewConstMetric(hardwareInfoDesc, prometheus.GaugeValue, float64(1), l...)
 	}
 }
+
+func (c *systemCollector) collectLicense(client *rpc.Client, ch chan<- prometheus.Metric, labelValues []string) {
+	r := &licenseInformation{}
+	err := client.RunCommandAndParse("show system license usage", r)
+
+	if err == nil {
+		for i := range r.LicenseInfo.License {
+			licenseLabels := append(labelValues,
+				strings.ToLower(r.LicenseInfo.License[i].Name),
+				strings.ToLower(r.LicenseInfo.License[i].Description),
+				strings.ToLower(r.LicenseInfo.License[i].ValidityType))
```

Comment:
Adding the expiry information to the label stack could be problematic.

It seems that this can also be a countdown in days or a timestamp (see [here](https://www.juniper.net/documentation/us/en/software/license/licensing/topics/topic-map/license-types.html#id-verifying-junos-os-license-installation-cli)):

`user@host> show system license usage
            Licenses     Licenses    Licenses    Expiry
  Feature name         used    installed      needed 
  subscriber-addr         1            0           1    29 days
  scale-subscriber        0         1000           0    permanent
  scale-l2tp              0         1000           0    permanent
  scale-mobile-ip         0         1000           0    permanent
`
In this case you would have a new label every day.

`user@host> show system license usage

                                   Licenses     Licenses    Licenses  Expiry
Feature name                           used    	installed      needed 
  av_key_kaspersky_engine               1            1           0    2012-03-30 01:00:00 IST
  wf_key_surfcontrol_cpa                0            1           0    2012-03-30 01:00:00 IST
  dynamic-vpn                           0            1           0    permanent
  ax411-wlan-ap                         0            2           0    permanent
`
With that you would get a new timeseries every time you update the license.

I think the only useful way to have this in there would be the countdown, but then not as a label but a as a separate metric.
From the timestamp you could also determine how many days that is until expiry.

With the countdown to expiry as an integer in a separate metric you could then see which licenses are about to expire in the next X days or are already expired (these should have negative values).

Don't know if that is worth the effort though. If not, I think I would leave it out if we don't want to alert on approaching license expirations.

Reply:
Great point. I also think that a countdown would make most sense here. 
We could have one time series for the expiry type ("regular", expired, invalid, permanent) as an Enum, then a second one with the actual countdown. 

Something like this, just in actual code:
`junos_system_license_expiry_type` = enum 0 ... 3
`junos_system_license_expiry_countdown` = `date(expiry) - date(now())`

Reply:
Implemented your idea with a slight adaptation:

```
0 ... n = Days until expiry
-1 = Expired
+Inf = Permanent license
-Inf = Invalid (whatever that is supposed to indicate
```

Reply:
Cool, that should work as well